test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/MockTracingTelemetry.java

```diff
@@ -46,6 +50,22 @@ public TracingContextPropagator getContextPropagator() {
 
     @Override
     public void close() {
+        shutdown();
+        spanProcessor.shutdown();
+        try {
```

Comment:
In case of OTel, this is the default behavior on `close()`, no need to duplicate that

```
  @Override
  default void close() {
    shutdown().join(10, TimeUnit.SECONDS);
  }
```

Reply:
https://github.com/opensearch-project/OpenSearch/pull/9415#discussion_r1305639677 Should implement in SpanProcessor then?

Reply:
Actually, some background monitoring action runs asynchronously and keep on creating the Spans. These threads are still active even after closing the Telemetry which is leading the StrictCheck to fail. So we need to wait for some time (10 seconds) to get these monitoring spans ended.

Reply:
Look, the node / cluster shutdown procedure is responsible for shutting those down, not the tracing framework. We also have thread leaks detector that catches those. So we either have a) incorrect shutdown procedure b) unnoticed thread leaks.

Reply:
`NodesStatsAction.java`
Seems, `InternalClusterInfoService` is running the scheduled tasks to refresh the cluster state and the refresh threads are only terminated on `ThreadPool.shutdown()`.

Reply:
Thank you, looking into it now.

Reply:
Could you please add tracer to Node's close method (after the thread pool):

```
public synchronized void close() throws IOException {
```

Reply:
It's already being closed after threadpool.shutdown https://github.com/opensearch-project/OpenSearch/blob/c90b6eac285055959d419ae67c369421355e78f8/server/src/main/java/org/opensearch/node/Node.java#L1513

But since, threadpool.shutdown() is not a sync and immediate operation which is causing the flakyness.

Reply:
Sorry didn't see that, have another idea, trying it out now.

Reply:
@Gaganjuneja could you please take a look at https://github.com/opensearch-project/OpenSearch/pull/9561, thank you

Reply:
Thanks @reta, I have approved the changes. Let me remove these changes from this PR and will take your changes before merge.

Reply:
Removed my changes and will rebase once your PR is merged.